File: frontend/src/types/export.ts
// types/export.ts
export interface ExportConfig {
  format: ExportFormat;          // CSV only (Phase 1)
  structure: ExportStructure;    // Wide or long
  includeConfidence: boolean;    // Include confidence scores?
  includeSourceText: boolean;    // Include source text spans?
  minConfidence?: number;        // Filter by minimum confidence (0-100)
}

export type ExportFormat = 'csv';  // Phase 2: 'excel' | 'json'

export type ExportStructure =
  | 'wide'   // One row per document (columns = variables)
  | 'long';  // One row per extracted value

export interface ExportResult {
  id: string;
  fileName: string;
  fileSize: number;
  createdAt: string;
  downloadUrl?: string;          // Available in Phase 1 (blob URL)
}

File: backend/src/schemas/export.py
"""
Pydantic schemas for export functionality.
"""
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field


class ExportFormat(str, Enum):
    """Export format enumeration."""
    CSV_WIDE = "CSV_WIDE"
    CSV_LONG = "CSV_LONG"
    EXCEL = "EXCEL"
    JSON = "JSON"


class ExportConfig(BaseModel):
    """
    Configuration for export generation.
    """
    format: ExportFormat = Field(..., description="Export format (CSV_WIDE, CSV_LONG, EXCEL, JSON)")
    include_confidence: bool = Field(default=False, description="Include confidence scores in export")
    include_source_text: bool = Field(default=False, description="Include source text excerpts in export")
    min_confidence: Optional[int] = Field(default=None, ge=0, le=100, description="Minimum confidence threshold (0-100)")


class ExportResponse(BaseModel):
    """
    Response for export generation.
    """
    download_url: str = Field(..., description="URL to download the export file")
    format: ExportFormat = Field(..., description="Export format")
    filename: str = Field(..., description="Generated filename")
    size_bytes: int = Field(..., description="File size in bytes")
